catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -319,23 +326,23 @@ public Object visit(PropertyIsEqualTo filter, Object delegate) {
     List<Object> functionArgs = filterValues.functionArgs;
 
     // Special case to handle relative temporal queries
-    if (literal instanceof String
-        && RELATIVE_TEMPORAL_MATCHER.matcher((String) literal).matches()) {
+    if (literal instanceof String && RELATIVE_TEMPORAL_REGEX.matcher((String) literal).matches()) {
       DateTime currentDateTime = new DateTime();
 
       // Split out the ISO 8601 duration from the ECQL function
-      String duration = (((String) literal).split("[\\(\\)]"))[1];
+      String periodText = (((String) literal).split("[\\(\\)]"))[1];
 
-      org.joda.time.Period period = org.joda.time.Period.parse(duration);
+      org.joda.time.Period period = PeriodParser.parse(periodText);
```

Comment:
Is there a reason we chose to make our own parser instead of a `PeriodFormatter` to hand to the `Period.parse` method?

Reply:
We could do that too but wouldn't that essentially do the same thing but hand back a string instead of a period?

Reply:
Looking at the source code for `Period.parse`, looks like it just calls and returns the value of `PeriodFormatter.parsePeriod`. According to the docs here (https://www.joda.org/joda-time/apidocs/org/joda/time/format/PeriodFormatter.html#parsePeriod-java.lang.String-) looks like that also returns a Period.

I think writing a `PeriodFormatter` and using Joda to call it still looks a little better, but I'm not sure if there are any other benefits to it, really. I meant to put a ✏️